src/hdmf_zarr/backend.py

```diff
@@ -789,24 +784,22 @@ def _create_ref(self, ref_object, export_source=None):
         """
         if isinstance(ref_object, Builder):
             if isinstance(ref_object, LinkBuilder):
-                builder = ref_object.target_builder
+                builder = ref_object.builder
             else:
                 builder = ref_object
         elif isinstance(ref_object, ReferenceBuilder):
             builder = ref_object.builder
-        else:
-            builder = self.manager.build(ref_object)
```

Comment:
If `ref_object` is a `Container`, which it seems like it can be based on line 635, then I think we should keep these lines. If it is not possible for ref_object to be a `Container`, then please update the if statement on line 635.